extensions/src/main/java/org/pentaho/platform/web/servlet/GenericServlet.java

```diff
@@ -81,6 +85,51 @@ public void init() throws ServletException {
     showDeprecationMessage = Boolean.parseBoolean( value );
   }
 
+  /**
+   * Determines if the request is for a static resource handled by this content generator handler
+   * assuming it's being served via its default URL, {@code /content}.
+   * <p>
+   * This particular method can be used without performing servlet initialization, via {@link #init()}.
+   * <p>
+   * The implementation of this method uses the exact same rules used by
+   * {@link #doGet(HttpServletRequest, HttpServletResponse)} to identify static resources. The two must be kept in sync.
+   *
+   * @param request The request.
+   * @return {@code true} if the request is for a static resource; {@code false}, otherwise.
+   */
+  public boolean isStaticResource( @NonNull final HttpServletRequest request ) {
+    if ( StringUtils.isEmpty( request.getPathInfo() ) ) {
+      return false;
+    }
+
+    String fullPath = request.getServletPath() + request.getPathInfo();
+
+    Matcher matcher = PATH_PATTERN.matcher( fullPath );
+    if ( !matcher.matches() ) {
+      return false;
+    }
+
+    // Assume it is a static resource, but then check the hypothesis.
+    String staticResourceUrl = matcher.group( 1 );
```

Comment:
📔  This variable has equivalent content to the `pathInfo` variable in the `doGet` method, if you want to compare the code (which this method should match).